### PR TITLE
Add telemetry based crash reporting for langserver

### PR DESCRIPTION
**Pull request by rasika**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29071

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by rasika** [comment]
Related PR https://github.com/ballerina-platform/plugin-vscode/pull/274

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSClientLogger.java [hunk 1]
```diff
@@ -83,16 +84,20 @@ public void notifyUser(String operation, Throwable error) {
      * @param identifier text document
      * @param pos        pos
      */
-    public void logError(String message, Throwable error, TextDocumentIdentifier identifier, Position... pos) {
+    public void logError(LSOperation operation, String message, Throwable error, TextDocumentIdentifier identifier,
+                         Position... pos) {
         if (!this.isInitializedOnce) {
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSClientLogger.java:89]: I think we can return early if language client is `null` and get rid of repetitive null checks.
```suggestion
        if (!this.isInitializedOnce || this.languageClient == null) {
```

**Review by IMS94 — APPROVED** [review]
@rasika Nice addition! LGTM except for a minor comment.

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**